docs/current_docs/configuration/modules.mdx

```diff
@@ -150,3 +150,19 @@ When a package manager is not explicitly defined within the `package.json` file,
 :::warning
 This behavior however should be considered a sensible fallback, and not as an explicit configuration. Since this default can change, we encourage you to configure a package manager explicitly.
 :::
+
+### Alternative base images
+
+You can specify a custom base image in your module's `package.json`, tailoring the SDK to your project's needs. To change the base image, set the field `dagger.baseImage` in your Dagger module's `package.json` file.
+
```

Comment:
This should be considered an escape hatch for advanced cases and come with a warning to "use at own risk". It's the only reason we're doing this, especially for users in private settings. Otherwise giving this much power to users could be a nightmare to support.

Most common use case would probably be adding some needed environment variables to the runtime container, adding a few files for authentication, or even needed operating system packages, for example. Users should be adivsed not to deviate from the default image too much, at risk of things breaking.

Ideally in the future, adding some features to the engine, would deprecate this. Python was the first to enable customizing the base image, but that came before the engine was even able to mount custom certificates to any container. Thus for that use case, it's no longer necessary to resort to this escape hatch.

As for adding OS packages, there's currently no other alternative, but the point is, ideally there will be. So again, best treat this as an escape hatch for advanced needs.

Reply:
Added